app/routes.py

```diff
@@ -1,2 +1,348 @@
+
+    response_body = {"task": task.to_dict()}
+    return jsonify(response_body), 200
+
+
+# *****  Wave 5  *********
+@goals_bp.route("", methods=["POST"])
+def goal_create():
+    request_body = request.get_json()
+    if "title" not in request_body:
+        return jsonify({"details": "Invalid data"}), 400
+
+    # collecting the value(s)
+    new_goal = Goal(title=request_body["title"])
+
+    # add and update database
+    db.session.add(new_goal)
+    db.session.commit()
+
+    # return the response to our client
+    response_body = {"goal": new_goal.dict()}
+    return jsonify(response_body), 201
+
+
+@goals_bp.route("", methods=["GET"])
+def handle_goals():
+    goals = Goal.query.all()
+    response_body = [goal.dict() for goal in goals]
+    return jsonify(response_body), 200
+
+
+@goals_bp.route("/<goal_id>", methods=["GET"])
+def goal_get(goal_id):
+    goal = Goal.query.get_or_404(goal_id)  # providing the task_id
+    response_body = {"goal": goal.dict()}
+    return jsonify(response_body), 200
+
+
+@goals_bp.route("/<goal_id>", methods=["PUT"])
+def goal_put(goal_id):
+    goal = Goal.query.get_or_404(goal_id)
+    form_data = request.get_json()
+
+    goal.title = form_data["title"]
+
+    db.session.commit()
+    response_body = {"goal": goal.dict()}
+    return jsonify(response_body), 200
+
+
+@goals_bp.route("/<goal_id>", methods=["DELETE"])
+def goal_delete(goal_id):
+    goal = Goal.query.get_or_404(goal_id)
+
+    db.session.delete(goal)
+    db.session.commit()
+
+    return jsonify({
+        "details": f"Goal {goal.goal_id} \"{goal.title}\" successfully deleted"
+    }), 200
+
+# *****  Wave 6  *********
+
+@goals_bp.route("/<goal_id>/tasks", methods=["POST"]) # do we use goals or goals_dp
+def post_goals_tasks(goal_id):
+    
+    goal = Goal.query.get(goal_id)
+
+    if goal is None:
+        return jsonify("Not Found"), 404
+    
+    request_body = request.get_json() # {"task_ids": [1, 2, 3]}
+    
+    # HOW to assign Tasks to goal_id
+    tasks_list = [] #class instances
+    for task_id in request_body["task_ids"]:
+        tasks_list.append(Task.query.get(task_id))
+        
+    goal.tasks = tasks_list
+    
+
+    # Need to update database after inserting values into Task/Goal Table?
+    # db.session.add()
+    db.session.commit()
+    
+    tasks_ids_list = [] #task id
+    for task in goal.tasks:
+        tasks_ids_list.append(task.task_id)
+
+    response_body = {
+        "id": goal.goal_id,
+        "task_ids": tasks_ids_list
+    }
+    
+    return jsonify(response_body), 200
+
+@goals_bp.route("/<goal_id>/tasks", methods=["GET"])
+def get_goals_tasks(goal_id):
+    goal = Goal.query.get_or_404(goal_id) #
+
+    tasks_list = []
+
+    for task in goal.tasks:
+        tasks_list.append({
+            "id": task.task_id,
+            "goal_id": goal.goal_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": task.completed_at is not None
+    })
```

Comment:
The `to_dict()` helper method could have been used here as it will add the `goal_id` key if it exists for the task. 
```suggestion
        tasks_list.append(task.to_dict())
```